src/views/MainHome.tsx

```diff
@@ -229,7 +229,12 @@ export default class MainHome extends Component<MainHomeProps, MainHomeState> {
             RoomComponent = <MainWelcome userDisplay={this.props.user} />;
         } else {
             if (!currentRoom.requestSent) {
-                RoomComponent = <RoomConnect sendRequests={this.sendRequests} />
+                RoomComponent = 
```

Comment:
If you like the ternary from above better, let's use that here too?
```
const roomComponent = currentRoom.requestSent ? roomConnectHtmlContent : mainWelcomeHtmlContent;
```

Reply:
Yep, will do but one drawback of this is that it still "loads" the components and if any of the `props` are undefined like `currentRoom.invited`, React will complain. But there are other way to go around that. Just wanted to point that out.

Reply:
Does "load" the component mean it also renders? Hmmmmm